Added OG Description Function to be called everytime an og description tag is set

rgharpuray opened the pull request:
I just added in the function for description, so everytime the description field is set, this function is called. 
Adding potential issue with parser to the Task, that may cause the wrong description to show up on the feed.

Commit:
added function for description for og tags

Commit:
Social publisher calls new description function

mattwkelly commented on fb-open-graph.php:
```diff
@@ -26,6 +26,31 @@ function fb_output_og_protocol( $property, $content ) {
 
+function fb_strip_and_format_desc($post) {
+
+	if ( !empty($excerpt_no_html) ) {
```
Will this ever be set ($excerpt_no_html)?

rgharpuray replied:
I dont think so, theres a check for it in line 31 but it doesnt seem to ever get set here. So I can probably just remove the if around the three lines of code, and have them always run.

rgharpuray replied:
actually, that whole 'if' including the three lines may not be needed anymore

mattwkelly commented on fb-social-publisher.php:
```diff
@@ -234,7 +235,7 @@ function fb_post_to_fb_page($post_id) {
         'name' => get_the_title(),
         'caption' => apply_filters( 'the_excerpt', get_the_excerpt() ),
-        'description' => apply_filters( 'the_excerpt', get_the_excerpt() ),
+        'description' => fb_strip_and_format_desc($post), //apply_filters( 'the_excerpt', get_the_excerpt() ),
```
is "//apply_filters( 'the_excerpt', get_the_excerpt() )," needed any more?  If not, remove it (including in the lines below).

mattwkelly replied:
Also, see http://codex.wordpress.org/WordPress_Coding_Standards

Make sure you have proper spaces by parentheses.

Commit:
fixes for first pull request

Commit:
Merge branch 'master' of github.com:mattwkelly/fb-hacks

mattwkelly commented:
Getting "Notice: Undefined index: show_on_homepage in /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/social-plugins/fb-like.php on line 47" -- see http://www.mattkel.ly/fb-hacks/wp/

You should use isset() or empty() (http://zainal.wordpress.com/2006/04/25/8/)

Commit:
changed acccess of options to use isset in fb-like.php in a few places

Commit:
added notification if user has potentially conflicting plugins with fb

rgharpuray commented:
I pushed to my repo (look at the fb-like.php, I added isset's there). More importantly though, check out fb-admin-menu.php  -- I added a function that checks if you have any plugins that might conflict with FB, and renders those toward the top of the fb admin menu. It's a static array you'll see in the fb_notify_user_of_plugin_conflicts function. I also added a link to the plugins page directly, so check that out. How's this look?

Commit:
removed new function, will add to new branch

mattwkelly commented:
http://www.mattkel.ly/fb-hacks/wp/?p=1344

Notice: Undefined variable: desc_no_html in /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/fb-open-graph.php on line 31 Call Stack: 0.0187 332972 1. {main}() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/index.php:0 0.0193 336408 2. require('/srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-blog-header.php') /srv/www/mattwkelly.com/public_html/fb-hacks/wp/index.php:17 0.7054 19471312 3. require_once('/srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/template-loader.php') /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-blog-header.php:16 0.7133 19480480 4. include('/srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/themes/twentyeleven/single.php') /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/template-loader.php:43 0.7133 19480480 5. get_header() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/themes/twentyeleven/single.php:10 0.7134 19480988 6. locate_template() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/general-template.php:34 0.7134 19481192 7. load_template() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/theme.php:1091 0.7139 19528376 8. require_once('/srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/themes/twentyeleven/header.php') /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/theme.php:1115 0.7378 19539460 9. wp_head() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/themes/twentyeleven/header.php:66 0.7378 19539536 10. do_action() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/general-template.php:1577 0.8184 19594224 11. call_user_func_array() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/plugin.php:405 0.8184 19594252 12. fb_add_og_protocol() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/plugin.php:405 0.8279 19595228 13. fb_strip_and_format_desc() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/fb-open-graph.php:80

mattwkelly commented:
If you haven't done so, I'd change your php config to show all notices.  http://php.net/manual/en/errorfunc.configuration.php

Commit:
initialized string for desc no html

Commit:
initialized string for desc no html
